<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yasin Kaymaz" />


<title>Data Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Harvard-RosenbrockLab</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="DataIntegration.html">DataIntegration</a>
</li>
<li>
  <a href="DimensionReduction_Clustering.html">Clustering</a>
</li>
<li>
  <a href="Contrasts.html">Contrasts</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yasinkaymaz/Harvard-RosenbrockLab.git">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data Analysis</h1>
<h4 class="author"><em>Yasin Kaymaz</em></h4>
<h4 class="date"><em>2/07/2019</em></h4>

</div>


<p><strong>Last updated:</strong> 2019-03-06</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20190304)</code> </summary></p>
<p>The command <code>set.seed(20190304)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/yasinkaymaz/Harvard-RosenbrockLab/tree/53e6dd49c5fdbfa527802e9cfd27ba3ce07fe715" target="_blank">53e6dd4</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/cell-annotate_rc.R

Unstaged changes:
    Modified:   code/WP_codes.R

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/yasinkaymaz/Harvard-RosenbrockLab/f4d06054fea80c1a3c4ab349295b36e18e54bae7/docs/DataIntegration.html" target="_blank">f4d0605</a>
</td>
<td style="text-align:left;">
Yasin Kaymaz
</td>
<td style="text-align:left;">
2019-03-05
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/yasinkaymaz/Harvard-RosenbrockLab/ced17eace8265f1157c883630f55e81bf2191946/docs/DataIntegration.html" target="_blank">ced17ea</a>
</td>
<td style="text-align:left;">
Yasin Kaymaz
</td>
<td style="text-align:left;">
2019-03-05
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/yasinkaymaz/Harvard-RosenbrockLab/c534f45cce704f6c1d9aacf281b06d838d90eece/docs/DataIntegration.html" target="_blank">c534f45</a>
</td>
<td style="text-align:left;">
Yasin Kaymaz
</td>
<td style="text-align:left;">
2019-03-04
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/yasinkaymaz/Harvard-RosenbrockLab/a1bf7eea3e9640238006c44042c86549abf9fa8d/docs/DataIntegration.html" target="_blank">a1bf7ee</a>
</td>
<td style="text-align:left;">
Yasin Kaymaz
</td>
<td style="text-align:left;">
2019-03-04
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/yasinkaymaz/Harvard-RosenbrockLab/blob/1be73852901af37590079568aff880ddd1a15621/analysis/DataIntegration.Rmd" target="_blank">1be7385</a>
</td>
<td style="text-align:left;">
Yasin Kaymaz
</td>
<td style="text-align:left;">
2019-03-04
</td>
<td style="text-align:left;">
initial commit
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="summary" class="section level1">
<h1>Summary</h1>
<!-- Single cell data from Tasic et al. 2018 is from two separate regions of mouse brain, Anterior Lateral Motor cortex (ALM) and Primary Visual cortex (VISp). Total single cells sequenced are 10,068 and 15,413, respectively. -->
<!-- The single cell data from Tasic et al. 2016 is total of 1,809 cells from Primary Visual cortex (VISp) of mouse brain. -->
</div>
<div id="main-steps" class="section level1">
<h1>Main Steps</h1>
<div id="hook-dataset" class="section level2">
<h2>Hook dataset</h2>
<p><strong>Citation:</strong> Hook, Paul W., Sarah A. McClymont, Gabrielle H. Cannon, William D. Law, A. Jennifer Morton, Loyal A. Goff, and Andrew S. McCallion. 2018. “Single-Cell RNA-Seq of Mouse Dopaminergic Neurons Informs Candidate Gene Selection for Sporadic Parkinson Disease.” American Journal of Human Genetics 102 (3): 427–46.</p>
<div id="obtain-the-data" class="section level3">
<h3>1. Obtain the data</h3>
<pre><code>473 single cell RNA-Seq samples from sorted mouse Th-eGFP+ dopaminergic neurons collected at two timepoints from three distinct brain regions.</code></pre>
<pre class="bash"><code>#wget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE108nnn/GSE108020/suppl/GSE108020_fpkm_table.txt.gz
#unzip GSE108020_fpkm_table.txt.gz</code></pre>
<p>After downloading the data, unzip the file of <strong>FPKM</strong> matrix for further analysis.</p>
<!-- ### 2. Filter out non-relevant genes prior to analysis. I took the filtration criteria from methods of Tasic et al., 2018. -->
<!-- Filter out -->
<!--   - Predicted gene models (gene names that start with Gm). -->
<!--   - Genes from mitochondrial chromosome. -->
<!--   - Ribosomal genes -->
<!--   - Sex-specific genes (Here is how I determined the gender-specific genes: [Code](https://github.com/yasinkaymaz/Harvard-NissenLab/tree/master/code/Determine_genes_to_filter.R)) -->
<!--   - Genes detected in less than 4 cells (min.cells = 4, Seurat) -->
</div>
<div id="filtration" class="section level3">
<h3>2. Filtration</h3>
<ul>
<li>Filter out cells expressing less than 500 genes (min.genes = 500, Seurat)</li>
</ul>
<!-- ## Tasic2016 dataset -->
<!-- ### 1. Obtain the data -->
<!-- ```{bash} -->
<!-- #wget https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE71585&format=file&file=GSE71585%5FClustering%5FResults%2Ecsv%2Egz -->
<!-- #wget https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE71585&format=file&file=GSE71585%5FRefSeq%5FTPM%2Ecsv%2Egz -->
<!-- ``` -->
<p>For the processing details, please follow <a href="https://github.com/yasinkaymaz/Harvard-GillardonLab/tree/master/code/WP_codes.R">Code</a></p>
<p>.</p>
<p>.</p>
<p>.</p>
<p>.</p>
<p>.</p>
<p>.</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-apple-darwin17.5.0 (64-bit)
Running under: macOS  10.14.3

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLAPACK.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.3.0   stringr_1.4.0   purrr_0.2.5     readr_1.1.1    
 [5] tidyr_0.8.1     tibble_2.0.1    tidyverse_1.2.1 dplyr_0.7.6    
 [9] Seurat_2.3.3    Matrix_1.2-14   cowplot_0.9.3   here_0.1       
[13] DT_0.4          plotly_4.8.0    ggplot2_3.1.0  

loaded via a namespace (and not attached):
  [1] diffusionMap_1.1-0   Rtsne_0.15           colorspace_1.4-0    
  [4] class_7.3-14         modeltools_0.2-22    ggridges_0.5.0      
  [7] mclust_5.4.1         rprojroot_1.3-2      htmlTable_1.12      
 [10] base64enc_0.1-3      rstudioapi_0.8       proxy_0.4-22        
 [13] flexmix_2.3-14       bit64_0.9-7          lubridate_1.7.4     
 [16] mvtnorm_1.0-8        xml2_1.2.0           codetools_0.2-15    
 [19] splines_3.5.0        R.methodsS3_1.7.1    robustbase_0.93-1   
 [22] knitr_1.20           Formula_1.2-3        jsonlite_1.6        
 [25] workflowr_1.1.1      broom_0.5.0          ica_1.0-2           
 [28] cluster_2.0.7-1      kernlab_0.9-26       png_0.1-7           
 [31] R.oo_1.22.0          compiler_3.5.0       httr_1.3.1          
 [34] backports_1.1.2      assertthat_0.2.0     lazyeval_0.2.1      
 [37] cli_1.0.1            lars_1.2             acepack_1.4.1       
 [40] htmltools_0.3.6      tools_3.5.0          bindrcpp_0.2.2      
 [43] igraph_1.2.1         gtable_0.2.0         glue_1.3.0          
 [46] reshape2_1.4.3       RANN_2.6             Rcpp_1.0.0          
 [49] cellranger_1.1.0     trimcluster_0.1-2    gdata_2.18.0        
 [52] ape_5.1              nlme_3.1-137         iterators_1.0.10    
 [55] fpc_2.1-11           lmtest_0.9-36        rvest_0.3.2         
 [58] irlba_2.3.2          gtools_3.8.1         DEoptimR_1.0-8      
 [61] zoo_1.8-3            MASS_7.3-50          scales_1.0.0        
 [64] hms_0.4.2            doSNOW_1.0.16        parallel_3.5.0      
 [67] RColorBrewer_1.1-2   yaml_2.2.0           reticulate_1.9      
 [70] pbapply_1.4-0        gridExtra_2.3        segmented_0.5-3.0   
 [73] rpart_4.1-13         latticeExtra_0.6-28  stringi_1.2.4       
 [76] foreach_1.4.4        checkmate_1.8.5      caTools_1.17.1      
 [79] SDMTools_1.1-221     rlang_0.3.1          pkgconfig_2.0.2     
 [82] dtw_1.20-1           prabclus_2.2-6       bitops_1.0-6        
 [85] evaluate_0.10.1      lattice_0.20-35      ROCR_1.0-7          
 [88] bindr_0.1.1          htmlwidgets_1.2      bit_1.1-14          
 [91] tidyselect_0.2.4     plyr_1.8.4           magrittr_1.5        
 [94] R6_2.3.0             snow_0.4-3           gplots_3.0.1        
 [97] Hmisc_4.1-1          haven_1.1.2          pillar_1.3.1        
[100] whisker_0.3-2        foreign_0.8-70       withr_2.1.2         
[103] mixtools_1.1.0       fitdistrplus_1.0-9   survival_2.42-6     
[106] scatterplot3d_0.3-41 nnet_7.3-12          tsne_0.1-3          
[109] modelr_0.1.2         crayon_1.3.4         hdf5r_1.0.1         
[112] KernSmooth_2.23-15   rmarkdown_1.10       readxl_1.1.0        
[115] grid_3.5.0           data.table_1.11.4    git2r_0.23.0        
[118] metap_0.9            digest_0.6.18        diptest_0.75-7      
[121] R.utils_2.6.0        stats4_3.5.0         munsell_0.5.0       
[124] viridisLite_0.3.0   </code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
