---
title: "Data Analysis"
author: "Yasin Kaymaz"
date: "2/07/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(plotly)
library(DT)
options(width = 80)
source("https://raw.githubusercontent.com/yasinkaymaz/cellRtools/master/main/functions.R")
```


# Main Steps for Hook dataset

  **Citation:** Hook, Paul W., Sarah A. McClymont, Gabrielle H. Cannon, William D. Law, A. Jennifer Morton, Loyal A. Goff, and Andrew S. McCallion. 2018. “Single-Cell RNA-Seq of Mouse Dopaminergic Neurons Informs Candidate Gene Selection for Sporadic Parkinson Disease.” American Journal of Human Genetics 102 (3): 427–46.


### 1. Obtain the data

	473 single cell RNA-Seq samples from sorted mouse Th-eGFP+ dopaminergic neurons collected at two timepoints from three distinct brain regions.
	
#### SRA raw fastq files:

  Using sratoolkit, downloaded raw fastq files from [SRA](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP126648)

#### Expression table deposited to GEO:

  For sanity check and quality control.
  
```{bash}
#wget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE108nnn/GSE108020/suppl/GSE108020_fpkm_table.txt.gz
#unzip GSE108020_fpkm_table.txt.gz
```

After downloading the data, unzip the file of **FPKM** matrix for further analysis.

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE, paged.print=FALSE}
gds <- GEOquery::getGEO(filename = '~/data/GSE108020_series_matrix.txt.gz')
```

### 2. Filtration

  - Filter out cells that are found to be low quality in the Hook et al. resulting in 396 cells for downstream analysis.
  
  - Filtered mitchondrial, ribosomal, and Gm-xxx genes as well as genes expressed less than 20 cells (same filtration applied in the paper).
  

## Gene & Isoform level expression

#### 1. Get reference data from Gencode

Reference mouse genome:
ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M20/GRCm38.primary_assembly.genome.fa.gz

Comprehensive gene annotation: 	ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M20/gencode.vM20.annotation.gtf.gz


#### Genome alignment of raw reads for expression quantifiation

Using Rsem-STAR pipeline, I aligned the reads to reference transcriptome and quantified isoforms (count, TPM).


For the processing details, please follow 
Gene-level: [Code](https://github.com/yasinkaymaz/Harvard-RosenbrockLab/tree/master/code/Rsem-Gene-counts.R)
Isoform-level: [Code](https://github.com/yasinkaymaz/Harvard-RosenbrockLab/tree/master/code/Rsem-Isoform-counts.R)


.

.

.

.

.

.
