---
title: "Isoform level Contrasts"
author: "Yasin Kaymaz"
date: "1/31/2019"
output:
  html_document:
    css: style.css
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(plotly)
library(DT)
library(tidyverse)
source("https://raw.githubusercontent.com/yasinkaymaz/cellRtools/master/main/functions.R")
source("code/functions.R")
Hook2018iso <- get(load("~/data/Hook2018iso.seurat.Robj"))

```

## Isoform expression distributions among cells in each group

Violin plots show normalized expression distribution of the isoform of the given gene in each cell (black dots) binned in various groups. **y-axis is in log2 scale!**

## Expression distributions based on **brain regions**

### **Gria1 isoforms**:


```{r echo=FALSE, fig.height=15, fig.width=20, message=FALSE, warning=FALSE}

Hook2018iso <- SetAllIdent(Hook2018iso, id = "region")

transcripts <- grep(pattern = "Gria1", x = rownames(Hook2018iso@data), value = TRUE)
VlnPlot(object = Hook2018iso, features.plot = transcripts, use.raw = TRUE, y.log = TRUE,nCol = 1,x.lab.rot = T)
```

### **Gria4 isoforms**:

```{r echo=FALSE, fig.height=20, fig.width=20, message=FALSE, warning=FALSE}

transcripts <- grep(pattern = "Gria4", x = rownames(Hook2018iso@data), value = TRUE)
VlnPlot(object = Hook2018iso, features.plot = transcripts, use.raw = TRUE, y.log = TRUE,nCol = 1,x.lab.rot = T)

```

### **Grm4 isoforms**:

```{r echo=FALSE, fig.height=30, fig.width=20, message=FALSE, warning=FALSE}

transcripts <- grep(pattern = "Grm4", x = rownames(Hook2018iso@data), value = TRUE)
VlnPlot(object = Hook2018iso, features.plot = transcripts, use.raw = TRUE, y.log = TRUE,nCol = 1,x.lab.rot = T)

```

### **Gpr83 isoforms**:

```{r echo=FALSE, fig.height=10, fig.width=20, message=FALSE, warning=FALSE}
transcripts <- grep(pattern = "Gpr83", x = rownames(Hook2018iso@data), value = TRUE)
VlnPlot(object = Hook2018iso, features.plot = transcripts, use.raw = TRUE, y.log = TRUE,nCol = 1,x.lab.rot = T)
```



## Expression distributions based on **cell subsets** identified in Hook et al.


### **Gria1 isoforms**:


```{r echo=FALSE, fig.height=15, fig.width=20, message=FALSE, warning=FALSE}

Hook2018iso <- SetAllIdent(Hook2018iso, id = "subset")

transcripts <- grep(pattern = "Gria1", x = rownames(Hook2018iso@data), value = TRUE)
VlnPlot(object = Hook2018iso, features.plot = transcripts, use.raw = TRUE, y.log = TRUE,nCol = 1,x.lab.rot = T)
```

### **Gria4 isoforms**:

```{r echo=FALSE, fig.height=20, fig.width=20, message=FALSE, warning=FALSE}

transcripts <- grep(pattern = "Gria4", x = rownames(Hook2018iso@data), value = TRUE)
VlnPlot(object = Hook2018iso, features.plot = transcripts, use.raw = TRUE, y.log = TRUE,nCol = 1,x.lab.rot = T)

```

### **Grm4 isoforms**:

```{r echo=FALSE, fig.height=30, fig.width=20, message=FALSE, warning=FALSE}

transcripts <- grep(pattern = "Grm4", x = rownames(Hook2018iso@data), value = TRUE)
VlnPlot(object = Hook2018iso, features.plot = transcripts, use.raw = TRUE, y.log = TRUE,nCol = 1,x.lab.rot = T)

```

### **Gpr83 isoforms**:

```{r echo=FALSE, fig.height=10, fig.width=20, message=FALSE, warning=FALSE}
transcripts <- grep(pattern = "Gpr83", x = rownames(Hook2018iso@data), value = TRUE)
VlnPlot(object = Hook2018iso, features.plot = transcripts, use.raw = TRUE, y.log = TRUE,nCol = 1,x.lab.rot = T)
```




.

.

.

.

.

.

